### ISSUE TITLE
Transpose Rows/Columns

### ISSUE THREAD

**Issue by mochsner** [tgrosinger/md-advanced-tables#22]
Would be quite useful to have this functionality to switch Rows and Columns (similar to Excel).

Coming to this library from the obsidian advanced tables plugin, which had a similar issue, but was closed as it would need to be developed here. That issue can be found here: https://github.com/tgrosinger/advanced-tables-obsidian/issues/73

If there's any guidance on where to get started with development for this, I'd potentially be interested as well. I'm sure the community would find this useful for everyone, especially if not sufficiently complicated.

**Comment by tgrosinger** [comment]
@mochsner, thanks for making this issue. I agree that it would be useful functionality in this library.

I think the way to get started with this would be creating a new function in [table-editor.ts](https://github.com/tgrosinger/md-advanced-tables/blob/main/src/table-editor.ts). You could take a look at the `sortRows` function in that file for an example. Similar to that function, you'll need to iterate through the contents of the table, build a new `allRows`, and return a new table. Once that is done we can hook it up to the table interface so it's callable.

**Comment by mochsner** [comment]
@tgrosinger Apologies for the delay on this one! New to yarn (somehow), and had a few things that kept me from starting until recently.

Regardless, I've got the core development for this functional (one unit test, and tested from Obsidian on Ubuntu 20.04), but the algorithm unfortunately isn't the cleanest in terms of readability without removing the delimiter row, and I'll probably add a few more tests / try to think of edge cases.

It seems this project is pretty dependent on the `AddRows` being used to create tables, so I just opted to use that directly as the initial approach, within the loops. Hopefully that's okay.

If performance ever becomes an issue (people are using crazy massive tables), I'm quite happy to revisit this. Or, maybe at some point I'll make another PR because it's bothering me. I have a few ideas for better speeds (adding new `Table()` functions that allow in-place modifications), but at this point, it seems to work pretty darn great from an efficiency perspective.

Otherwise, I'm mostly just curious your take on these things below, focused on readability, with 2 different "paths" I can go down and knock 'em out.

Option 1: Removing the delimiter row before calculations
    - This would mean there's no no `iter+1`'s to on `.getCellAt`).
    - 1-expanded: **Adding "AutoAddDelimiterRow" logic to the `Table()` constructor**
      - skipping the x === 1 logic in this case on the outer loop, as well as any 
      
Option 2: Renaming variables with +1 counters and general counters to just be a bit more intuitive (or maybe use w instead of x, and h instead of y) -- that sort of change.

Alternatively, the code in my repository is technically fully functional, so if you're eager to get this into Obsidian for starters, I can go ahead and start on development there -- designing the SVG icon seems like the most of the work there, though.
